### ISSUE TITLE
No performance different between ORT_ENABLE_ALL and ORT_DISABLE_ALL 

### ISSUE THREAD

**Issue by sapjunior** [microsoft/onnxruntime#4816]
**Describe the bug**

No performance gain at all compared between ORT_ENABLE_ALL and ORT_DISABLE_ALL
Link to model: https://drive.google.com/file/d/14Lv1dDiOU9B_FeKe7_6mkh8j_XpnlKZg/view?usp=sharing

Here is the speed comparison code

```
import onnxruntime as rt
import numpy as np
import time

sessOptionsA = rt.SessionOptions()
sessOptionsA.graph_optimization_level = rt.GraphOptimizationLevel.ORT_ENABLE_ALL    
rtA = rt.InferenceSession('t10-exp.onnx', sessOptionsA)

sessOptionsB = rt.SessionOptions()
sessOptionsB.graph_optimization_level = rt.GraphOptimizationLevel.ORT_DISABLE_ALL    
rtB = rt.InferenceSession('t10-exp.onnx', sessOptionsB)

inputNodeName = rtA.get_inputs()[0].name
dummyData = np.zeros((1,3,112,112),dtype=np.float32)


avgTimeA = 0
for i in range(20):
    st = time.time()
    rtA.run([], {inputNodeName: dummyData})[0]
    en= time.time()
    avgTimeA+=(en-st)


avgTimeB = 0
for i in range(20):
    st = time.time()
    rtB.run([], {inputNodeName: dummyData})[0]
    en= time.time()
    avgTimeB+=(en-st)


print('ENABLE',avgTimeA/20,'DISABLE',avgTimeB/20)
```

`ENABLE 0.08494850397109985 DISABLE 0.08426977396011352`

**Comment by hariharans29** [comment]
Not sure if ORT_ENABLE_ALL ***guarantees*** perf gain when compared to ORT_DISABLE_ALL. ORT_ENABLE_ALL tries to enable optimizations in the model if applicable. But, if the model is already optimized and there is nothing left for ORT_ENABLE_ALL, then it is expected to have no diff between enabling all optimizations and disabling all optimizations isn't it ?

The correct comparison should be between the base framework perf and ORT perf after ORT_ENABLE_ALL. If you see a diff here, that warrants investigation. I don't think not seeing a diff between ORT_DISABLE_ALL and ORT_ENABLE_ALL is an issue (it may happen sometimes by the above logic)

**Comment by tracysh** [comment]
Hi, I took a look at the model and have a few comments:

1. The different optimizations levels aren't having an impact because the model appears to have been exported with keep_initializers_as_inputs=True through torch.onnx.export. This signals to the runtime that any of the weights may be overriden at inference time, so none of the optimizations apply. I see this debug message when using the model with onnxruntime_perf_test:

2020-08-17 08:49:14.6667507 [W:onnxruntime:, graph.cc:863 onnxruntime::Graph::Graph] Initializer 90 appears in graph inp
uts and will not be treated as constant value/weight. This may prevent some of the graph optimizations, like const foldi
ng. Move it out of graph inputs if there is no need to override it, by either re-generating the model with latest export
er/converter or with the tool onnxruntime/tools/python/remove_initializer_from_input.py.

2. I edited the model to remove the weights from the inputs list. This then allowed various optimizations such as Conv+BatchNorm fusion and NCHWc layout transformation to occur. However, the overall performance ended up much worse. I'm seeing that the weights contain denormal numbers which results in a very slow path in the processor (an order of magnitude worse for a multiply/add instruction).

I'm curious if you are also seeing a slowdown when doing an inference with PyTorch. PyTorch forums point users at torch.set_flush_denormal(true) to enable a CPU mode that improves performance by returning zero instead of attempting to compute the correct denormal. ONNX Runtime does not have an equivalent for that today. I modified the runtime to force this mode on all the threads and then performance improved greatly: ~400ms->150ms. I see the same behavior using the Intel DNNL execution provider.

If the weights could be post-processed to clip very small numbers, that should avoid the denormal problem. I've just started seeing more models with this problem recently: I don't have a plan at this time of how the runtime might have an equivalent to set_flush_denormal (or go the TensorFlow and always flush denormals to zero).
